### ISSUE TITLE
Allow CSV to define PodSecurityPolicies (or Openshift SecurityContextConstraints)

### ISSUE THREAD

**Issue by RyuunoAelia** [operator-framework/operator-lifecycle-manager#1547]
## Feature Request

**Is your feature request related to a problem? Please describe.**
Some platforms (and probably more and more) are restricting pods to avoid exiting the containers. A good example of that is OpenShift. In that case, some Operators might need to define custom security rules because the ones provided by the platform are too restrictive.
The current status is that each operator will have to define a way to do it by either:

* asking the user to do it as a prerequisite (error prone, a hassle if there are updates to the rights for each version, etc)
* requesting the rights in RBAC to do it itself
* having another operator doing it beforehand (this is actually what Red Hat chose to do in OpenShift Container Storage : https://github.com/openshift/ocs-operator/blob/master/pkg/controller/ocsinitialization/sccs.go )

The problem here is that there is no standard way to do it, and, specially for the two last solutions, there is no easy way to understand the real rights needed by the operator (without even considering the possibility of bug that could break pre-defined PSP/SCC in the cluster).

**Describe the solution you'd like**
Having a section of the CSV defining the PSP/SCC needed by the Operator, there would be the following advantages:

* an easy way to review the actions the operator is able to perform.
* no need for an external setup step on clusters that have these security features in enforcing mode.
* standardise the ecosystem, which is a benefit for the community as a whole.

**Comment by exdx** [comment]
We are working to add support for PSPs in the operator bundle in 4.6 - see #1564

**Comment by RyuunoAelia** [comment]
This is not exactly what I see in the PR pointed here. I do see that it is a design doc for adding new objects in the CSV, but it specifically mentions 3 Kinds and not PSP or SCC.

**Comment by exdx** [comment]
> This is not exactly what I see in the PR pointed here. I do see that it is a design doc for adding new objects in the CSV, but it specifically mentions 3 Kinds and not PSP or SCC.

That's correct, my mistake. PSP was something else we looked at but is not included in that particular design. However, we are looking to add support for all arbitrary kube objects in the bundle in the upcoming releases of OLM - we know this is an important feature for the community.

**Comment by RyuunoAelia** [comment]
If we are playing spot the differencies between openshift and kubernetes we have a lot to point.
* SCC vs PSP
* Route vs Ingress
* (at least for openshift 3.x) RBAC resources are in apiVersion authorization.openshift.io/v1  versus rbac.authorization.k8s.io/v1

probably other stuff I don't know about.

I think that simply saying that objects are in the bundle and OLM will create them as-is poses a lot of issues for compatibility between different versions of OpenShift and kubernetes.

This goes farther than simply the differences between OpenShift and kubernetes, but also the apiVersion changes between versions of kubernetes.

I also see another big issue for naming, as objects like PSP/SCC are cluster-wide, there is a risk of name collision.
Changing the name of PSP/SCC by putting a random string like it is currently done for ClusterRoles, implies that OLM fully understands the usages of the object and controls the references of each usage.

**Comment by cdjohnson** [comment]
bump
